docs/rfcs/1013-parsec-auth-scheme.md

```diff
@@ -0,0 +1,267 @@
+
+```yml
+{
+  "cmd": "account_creation_step1_email_validation",
+  "req": {
+    "email": "Email"
+  },
+  "reps": [
+    {
+      "status": "ok"
+    },
+    {
+      "status": "invalid_email"
+    }
+  ]
+}
+```
+
+On `ok`, the server would have sent a mail with a unique token used for next the request used to register the authentication method:
+
+```yml
+{
+  "cmd": "account_creation_step2_registration",
+  "req": {
+    "email_validation_token": "EmailValidationToken",
+    "account_manifest": "EncryptedAccountManifest",
+    "auth_medium": "AuthMedium"
+  },
+  "reps": [
+    {
+      "status": "ok",
+      "auth_medium_id": "AuthMediumID"
+    },
+    {
+      "status": "invalid_email_validation_token"
+    }
+  ],
+  "custom_types": {
+    "AuthMedium": {
+      "id": "AuthMediumID",
+      "account_manifest_encrypted_sym_key": "EncryptedSymKey",
+      "hmac_key": "HMACKey",
+      "encrypted_priv_key": "EncryptedPrivKey",
+      "encrypted_priv_key_key_algorithm": "KeyAlgorithm"
+    }
+  }
+}
+```
+
+`EncryptedAccountManifest`, `EncryptedSymKey`, `HMACKey`, `EncryptedPrivKey` are specialized `Bytes` types.
+
+`EncryptedAccountManifest` is generated by encrypting the following rust structure:
+
+```rust
+struct AccountManifest {
+  /// List of registered auth medium for the account.
+  auth_mediums: HashMap<AuthMediumID, PublicKey>,
+  /// List of devices registered for the account.
+  device_keys: HashMap<(OrganizationID, DeviceID), SymmetricKey>,
+}
+```
+
+### Obtain the account manifest
+
+Before obtaining the encrypted account manifest, the user retrieves the information related to its authentication medium:
+
+```yml
+{
+  "cmd": "auth_medium_get",
+  "req": {
+    "id": "AuthMediumID"
+  },
+  "reps": [
+    {
+      "status": "ok",
+      "account_manifest": "EncryptedAccountManifest",
+      "encrypted_priv_key_key_algorithm": "KeyAlgorithm",
+      "encrypted_priv_key": "EncryptedPrivKey"
+    }
+  ]
+}
+```
+
+And now request the account manifest:
+
+```yml
+{
+  "cmd": "account_manifest_get",
+  "req": {},
+  "status": [
+    {
+      "status": "ok",
+      "account_manifest": "EncryptedAccountManifest"
+    }
+  ]
+}
+```
+
+### Uploading a new device
+
+Now that the client can encrypt the device with the symmetric key, it can upload it:
+
+```yml
+{
+  "cmd": "device_upload",
+  "req": {
+    "account_manifest": "EncryptedAccountManifest",
+    "organization_id": "OrganizationID",
+    "device_ownership_proof": "DeviceOwnershipProof",
+    "encrypted_device": "EncryptedDevice"
+  },
+  "reps": [
+    {
+      "status": "ok"
+    },
+    {
+      "status": "auth_medium_mismatch"
+    },
+    {
+      "status": "invalid_ownership_proof"
+    }
+  ]
+}
+```
+
+> [!NOTE]
+> The user can upload multiple devices for the same organization.
+
+### List available devices
+
+To list the devices registered in the service, the client only needs to be authenticated:
```

Comment:
This route is no longer useful since those data should be store in the account manifest

Reply:
All the encrypted devices would be stored in the account manifest? So is the client that directly stores that in the manifest?

If that's the case there is no need to "list" or "get" devices, but there is also no need to "upload" them either??

Reply:
We still need the list for the `revoked_on` field, current the manifest only store symmetric key to decrypt the devices.

Should we change that?